los-deferal-02/Hello-Books
# #166238077 User should be able to request for book

#### What does this PR do?
Implements the feature for user to request for a book that is not yet in the library

#### Description of Task to be completed?
- Create models, migrations, and controllers for this feature
- Validate the inputs from the user
- Write integrated tests for the feature
- Document the feature's API using Swagger

#### How should this be manually tested?
On Terminal:

- Run git clone https://github.com/los-deferal-02/Hello-Books.git to clone this repo onto your local machine
- Navigate to the repo directory by running cd Hello-Books/
- Ensure your working directory is clean by running git status command
- Get a copy of the PR by typing git fetch origin pull/23/head:ft-create-book-request-166238077
- Now that you have a copy of the branch, switch to it using git checkout ft-create-book-request-166238077
- Your directory will now be an exact copy of the PR
- Run npm install to install all project's dependencies onto your local machine

#### Any background context you want to provide?
It is important that a user should be able to request for a book that does not currently exist in the library. This will advise management on the books to invest in.

#### What are the relevant pivotal tracker stories?
[#166238077](https://www.pivotaltracker.com/story/show/166238077)

#### Screenshots (if appropriate)
![book-request-pr-scrnshot](https://user-images.githubusercontent.com/32547932/59723755-6ce37c00-921f-11e9-8038-37c6cb648af0.png)
